spharpy/spherical.py

```diff
@@ -1,104 +1,585 @@
+            raise ValueError(
+                "Gradient computation not supported for MaxN, "
+                "SN3D normalization or FuMa channel ordering"
+            )
+        else:
+            if self.basis_type == "complex":
+                grad_theta, grad_phi = spherical_harmonic_basis_gradient(
+                    self.n_max, self.coords
+                )
+                self.basis_gradient_theta, self.basis_gradient_phi = (
+                    grad_theta,
+                    grad_phi,
+                )
+            elif self.basis_type == "real":
+                grad_theta, grad_phi = spherical_harmonic_basis_gradient_real(
+                    self.n_max, self.coords
+                )
+                self.basis_gradient_theta, self.basis_gradient_phi = (
+                    grad_theta,
+                    grad_phi,
+                )
+            else:
+                raise ValueError(
+                    "Invalid signal type, should be either 'complex' or 'real'"
+                )
+
+    def compute_basis_gradient(self):
+        try:
+            return self._compute_basis_gradient()
+        except Exception as e:
+            raise ValueError("Error computing basis gradient:", e) from e
+
+    @property
+    def basis_inv(self):
+        if self._basis_inv is None:
+            self.compute_inverse()
+        self._check_properties(compute='basis_inv')
+        return self._basis_inv
+
+    def compute_inverse(self):
+        """
+        Compute the inverse transform matrix for the specified transform type
+
+        The inverse transform matrix is calculated based on the specified
+        `inverse_transform`. If ``'pseudo_inverse' is chosen,
+        the Moore-Penrose pseudo-inverse is used.
+        If ``'quadrature'`` is chosen, the inverse is computed
+        using the conjugate transpose of the basis matrix
+        multiplied by 4 * pi * weights.
+
+        Returns
+        -------
+        None
+        """
+        if self.basis is None:
+            self.compute_basis()
+        if self.inverse_transform is not None:
+            assert self.inverse_transform in ["pseudo_inverse",
+                                              "quadrature"], (
+                "Invalid inverse transform type, "
+                "currently only 'pseudo_inverse' "
+                "and 'quadrature' are supported"
+            )
+        # print("computing inverse basis using ", self.inverse_transform)
+        elif self.inverse_transform is None:
+            ValueError("Inverse transform type not specified")
+        if self.inverse_transform == "pseudo_inverse":
+            self._basis_inv = np.linalg.pinv(self.basis)
+        elif self.inverse_transform == "quadrature":
+            if self.weights is None:
+                print(
+                    "Warning: No weights specified for quadrature transform,"
+                    "calculating weights using voronoi tessellation of sphere"
+                )
```

Comment:
```suggestion
                warnings.warn(
                    "No weights specified for quadrature transform, "
                    "calculating weights using voronoi tessellation of sphere"
                )
```
make sure to import the warnings module :)

Reply:
@f-brinkmann The problem is that you can also calculate the spherical harmonics inverse using regularised least squares (or another way possibly) and that may be a feature for a next version.